main.tf

```diff
@@ -12,6 +13,25 @@ terraform {
   backend "inmem" {}
 }
 
+provider "imagetest" {
+  harnesses = {
```

Comment:
as written, this approach adds the necessary provider config to the `main.tf` to make things work for CI presubmit, as well as for folks using `k3d` locally.

however, this will **fail** for local dev when `k3d-k3s-default` is not used, which isn't the best UX.

an alternative would be to create this in a `main_override.tf` within the presubmit job so that this only applies to presubmit CI.